rackspace/metrics/v2/ingestion/requests.go

```diff
@@ -0,0 +1,205 @@
+package ingestion
+
+import (
+	"github.com/rackspace/gophercloud"
+	"fmt"
+)
+
+// SendMetrics will ingest the metrics for the tenant associated with RackspaceMetrics.
+func SendMetrics(c *gophercloud.ServiceClient, metrics []MetricData) {
+	var res PostResult
+	reqBody := make([]map[string]interface{}, len(metrics))
+	for i := range metrics {
+		Metric := metrics[i]
+		reqBody[i] = map[string]interface{}{
+			"collectionTime": Metric.CollectionTime,
+			"ttlInSeconds":Metric.TtlInSeconds,
+			"metricValue": Metric.MetricValue,
+			"metricName": Metric.MetricName,
```

Comment:
One more question: concerning this code:
```go
reqBody[i] = map[string]interface{}{
	"collectionTime": Metric.CollectionTime,
	"ttlInSeconds":Metric.TtlInSeconds,
	"metricValue": Metric.MetricValue,
	"metricName": Metric.MetricName,
...etc...
```
Will this collection of JSON attributes change from within Gophercloud?  E.g., will there be protocol extensions which Gophercloud has to deal with that fundamentally alters which attributes appear inside this JSON object?

**If not**, then I'd agree that you should replace this with a `struct` with some JSON encoding attributes.  Look in the encoding/json package documentation for more details on how that's done.

**If so,** then I'm sorry @rgbkrk, but there's no hope.  It'll have to stay a map of interfaces because there's no way to statically predefine variant records in a JSON-encodable manner.  This is why I wanted to know more about the code.  Lots of variables and unknowns at this point.